src/pwncore/routes/ctf/pre_event.py

```diff
@@ -28,25 +36,57 @@ async def ctf_list():
     return problems
 
 
+@router.get("/today")
+async def ctf_today():
+    return await PreEventProblem_Pydantic.from_queryset(
+        PreEventProblem().filter(date=datetime.now(_IST).date())
+    )
+
+
+@router.get("/coins")
+async def coins_get(cq: CoinsQuery):
+    try:
+        return (
+            await PreEventUser.get(tag=cq.tag.strip().casefold())
+            .annotate(coins=Sum("solvedproblems__problem__points"))
+            .values_list("coins", flat=True)
+        )
+    except DoesNotExist:
+        return 0
+
+
 @atomic()
 @router.post("/{ctf_id}/flag")
 async def pre_event_flag_post(ctf_id: int, post_body: PreEventFlag, response: Response):
     problem = await PreEventProblem.get_or_none(id=ctf_id)
+
     if not problem:
         response.status_code = 404
         return {"msg_code": config.msg_codes["ctf_not_found"]}
 
     user_tag = post_body.tag.strip().casefold()
 
-    if await PreEventSolvedProblem.exists(tag=user_tag, problem_id=ctf_id):
+    if await PreEventSolvedProblem.exists(user_id=user_tag, problem_id=ctf_id):
```

Comment:
Shouldn't it be `user_tag` instead of `user_id` or does Tortoise assume the ID is the pk of the model?